<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1058</width>
    <height>1079</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>200</width>
    <height>300</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Registration Tool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <item>
     <widget class="QFrame" name="frame_principal">
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>400</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(181, 181, 181);</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="0">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="main_layout" stretch="1,3">
         <item>
          <widget class="QFrame" name="frame_up">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,1,1">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QFrame" name="frame_button_image">
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QGridLayout" name="gridLayout" columnstretch="2,1,1,2">
               <item row="2" column="3">
                <widget class="QPushButton" name="pushButton_open_mov_hypercube">
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(150, 150, 150);</string>
                 </property>
                 <property name="text">
                  <string>Open Moving Cubes</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QPushButton" name="pushButton_open_ref_hypercube">
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(150, 150, 150);</string>
                 </property>
                 <property name="text">
                  <string>Open Reference Cubes</string>
                 </property>
                 <property name="autoExclusive">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item row="3" column="0" colspan="2">
                <widget class="QGroupBox" name="groupBox_image_ref">
                 <property name="title">
                  <string>Gray Mode for Ref</string>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,1,1">
                  <item>
                   <widget class="QRadioButton" name="radioButton_whole_ref">
                    <property name="enabled">
                     <bool>true</bool>
                    </property>
                    <property name="text">
                     <string>Whole range</string>
                    </property>
                    <property name="checked">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QRadioButton" name="radioButton_one_ref">
                    <property name="text">
                     <string>One channel</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_ref_chan">
                    <item>
                     <widget class="QSlider" name="horizontalSlider_ref_channel">
                      <property name="maximum">
                       <number>149</number>
                      </property>
                      <property name="value">
                       <number>0</number>
                      </property>
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="spinBox_ref_channel">
                      <property name="maximum">
                       <number>149</number>
                      </property>
                      <property name="singleStep">
                       <number>1</number>
                      </property>
                      <property name="value">
                       <number>0</number>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </widget>
               </item>
               <item row="3" column="2" colspan="2">
                <widget class="QGroupBox" name="groupBox_imag_move">
                 <property name="title">
                  <string>Gray Mode for Moving</string>
                 </property>
                 <layout class="QVBoxLayout" name="verticalLayout_2">
                  <item>
                   <widget class="QRadioButton" name="radioButton_whole_mov">
                    <property name="enabled">
                     <bool>true</bool>
                    </property>
                    <property name="text">
                     <string>Whole range</string>
                    </property>
                    <property name="checked">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QRadioButton" name="radioButton_one_mov">
                    <property name="text">
                     <string>One channel</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <layout class="QHBoxLayout" name="horizontalLayout_mov_chan">
                    <item>
                     <widget class="QSlider" name="horizontalSlider_mov_channel">
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QSpinBox" name="spinBox_mov_channel"/>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </widget>
               </item>
               <item row="2" column="1" colspan="2">
                <widget class="QPushButton" name="pushButton_switch_images">
                 <property name="text">
                  <string> &lt;-&gt;</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="frame_spectra">
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0">
               <item>
                <widget class="QGroupBox" name="groupBox">
                 <property name="title">
                  <string>Registration parameters</string>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_3" columnstretch="1,0">
                  <item row="2" column="0">
                   <spacer name="verticalSpacer_2">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item row="10" column="1">
                   <widget class="QCheckBox" name="checkBox_autorize_modify">
                    <property name="text">
                     <string>Autorize modifying registered cube</string>
                    </property>
                    <property name="checked">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="0">
                   <widget class="QLabel" name="label">
                    <property name="text">
                     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Matching Method&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QLabel" name="label_2">
                    <property name="text">
                     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Transformation&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                    </property>
                   </widget>
                  </item>
                  <item row="10" column="0">
                   <widget class="QCheckBox" name="checkBox_crop">
                    <property name="text">
                     <string>Only selected zone ?</string>
                    </property>
                   </widget>
                  </item>
                  <item row="12" column="1">
                   <widget class="QPushButton" name="pushButton_getFeatures">
                    <property name="styleSheet">
                     <string notr="true">background-color: rgb(150, 150, 150);</string>
                    </property>
                    <property name="text">
                     <string>Get features</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QComboBox" name="method_selector">
                    <property name="currentText">
                     <string notr="true">AKAZE</string>
                    </property>
                    <item>
                     <property name="text">
                      <string>AKAZE</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>ORB</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QComboBox" name="transform_selector">
                    <property name="currentText">
                     <string>Perspective</string>
                    </property>
                    <item>
                     <property name="text">
                      <string>Perspective</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Affine</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item row="12" column="0">
                   <widget class="QPushButton" name="pushButton_register">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                    <property name="styleSheet">
                     <string notr="true"/>
                    </property>
                    <property name="text">
                     <string>Re-register</string>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="0" colspan="2">
                   <widget class="QGroupBox" name="horizontalGroupBox">
                    <layout class="QHBoxLayout" name="horizontalLayout_4">
                     <item>
                      <widget class="QLabel" name="label_feature">
                       <property name="text">
                        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;right&quot;&gt;% features to keep :
&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                       </property>
                       <property name="alignment">
                        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                       </property>
                      </widget>
                     </item>
                     <item>
                      <widget class="QSlider" name="features_slider">
                       <property name="maximum">
                        <number>100</number>
                       </property>
                       <property name="pageStep">
                        <number>1</number>
                       </property>
                       <property name="value">
                        <number>50</number>
                       </property>
                       <property name="orientation">
                        <enum>Qt::Horizontal</enum>
                       </property>
                      </widget>
                     </item>
                    </layout>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QFrame" name="frame_2">
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <widget class="QGroupBox" name="groupBox_Visualization">
                 <property name="title">
                  <string>Visualization</string>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,1">
                  <item row="1" column="1">
                   <widget class="QSlider" name="horizontalSlider_keyPacketToShow">
                    <property name="pageStep">
                     <number>1</number>
                    </property>
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="tickPosition">
                     <enum>QSlider::NoTicks</enum>
                    </property>
                   </widget>
                  </item>
                  <item row="3" column="1">
                   <widget class="QComboBox" name="overlay_selector">
                    <item>
                     <property name="text">
                      <string>Checkboard</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Color</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>View Matches</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>Only aligned</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                  <item row="3" column="0">
                   <widget class="QLabel" name="label_5">
                    <property name="text">
                     <string>Results view :</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="1">
                   <spacer name="verticalSpacer">
                    <property name="orientation">
                     <enum>Qt::Vertical</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>20</width>
                      <height>40</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item row="0" column="0">
                   <widget class="QLabel" name="label_3">
                    <property name="text">
                     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;features per packet&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QSpinBox" name="spinBox_keypointPerPacket">
                    <property name="wrapping">
                     <bool>false</bool>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignCenter</set>
                    </property>
                    <property name="maximum">
                     <number>100</number>
                    </property>
                    <property name="value">
                     <number>10</number>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QLabel" name="label_packetToShow">
                    <property name="text">
                     <string>features</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item row="5" column="1">
                   <widget class="QPushButton" name="pushButton_save_cube">
                    <property name="enabled">
                     <bool>false</bool>
                    </property>
                    <property name="styleSheet">
                     <string notr="true">background-color: rgb(150, 150, 150);</string>
                    </property>
                    <property name="text">
                     <string>Save Registered Cube</string>
                    </property>
                    <property name="autoExclusive">
                     <bool>false</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="5" column="0">
                   <widget class="QPushButton" name="pushButton_reset">
                    <property name="styleSheet">
                     <string notr="true">color: rgb(255, 0, 0);</string>
                    </property>
                    <property name="text">
                     <string>Reset All</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="frame_down">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QSplitter" name="splitter">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <widget class="QWidget" name="layoutWidget">
               <layout class="QVBoxLayout" name="left_layout">
                <property name="sizeConstraint">
                 <enum>QLayout::SetDefaultConstraint</enum>
                </property>
               </layout>
              </widget>
              <widget class="QWidget" name="layoutWidget">
               <layout class="QVBoxLayout" name="right_layout"/>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>pushButton_open_ref_hypercube</tabstop>
  <tabstop>pushButton_open_mov_hypercube</tabstop>
  <tabstop>radioButton_whole_ref</tabstop>
  <tabstop>radioButton_one_ref</tabstop>
  <tabstop>horizontalSlider_ref_channel</tabstop>
  <tabstop>spinBox_ref_channel</tabstop>
  <tabstop>radioButton_whole_mov</tabstop>
  <tabstop>radioButton_one_mov</tabstop>
  <tabstop>horizontalSlider_mov_channel</tabstop>
  <tabstop>spinBox_mov_channel</tabstop>
  <tabstop>transform_selector</tabstop>
  <tabstop>method_selector</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>horizontalSlider_ref_channel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_ref_channel</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>197</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_ref_channel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_ref_channel</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>115</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_mov_channel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_mov_channel</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>252</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>303</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_mov_channel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_mov_channel</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>258</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
